Exiled.Events/Patches/Events/Map/ChangingIntoGrenade.cs

```diff
@@ -0,0 +1,99 @@
+// -----------------------------------------------------------------------
+// <copyright file="ChangingIntoGrenade.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Map
+{
+#pragma warning disable SA1118
+    using System.Collections.Generic;
+    using System.Reflection;
+    using System.Reflection.Emit;
+
+    using Exiled.API.Features;
+    using Exiled.Events.EventArgs;
+
+    using Grenades;
+
+    using HarmonyLib;
+
+    using Mirror;
+
+    using NorthwoodLib.Pools;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="FragGrenade.ChangeIntoGrenade"/>.
+    /// Adds the <see cref="Handlers.Map.ChangingIntoGrenade"/> event.
+    /// </summary>
+    [HarmonyPatch(typeof(FragGrenade), nameof(FragGrenade.ChangeIntoGrenade))]
+    internal static class ChangingIntoGrenade
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            // The index offset
+            int offset = 2;
+
+            // Find the last return false call.
+            int index = newInstructions.FindLastIndex(instruction => instruction.opcode == OpCodes.Ldc_I4_0) + offset;
+
+            // Generate a continue lable.
+            Label continueLabel = generator.DefineLabel();
+
+            // Declare ServerChangingGrenadeEventArgs, to be able to store it's instance with "stloc.s".
```

Comment:
You should change the comment to reflect the new name (`ChangingGrenadeEventArgs`).